ImageMagick uses many unsupported attributes by Oracle Solaris Studio.
Supported attributes are here:
http://docs.oracle.com/cd/E24457_01/html/E21991/gljol.html
This patch comes from in-house; as it is Solaris-specific, it will not
be contributed upstream.

--- ImageMagick-6.8.3/magick/method-attribute.h.~1~	2013-01-16 14:17:00.000000000 -0800
+++ ImageMagick-6.8.3/magick/method-attribute.h	2015-02-10 20:34:01.057456551 -0800
@@ -110,7 +110,7 @@
 
 #if defined(MAGICKCORE_HAVE___ATTRIBUTE__)
 #  define magick_aligned(x)  __attribute__((aligned(x)))
-#  define magick_attribute  __attribute__
+#  define magick_attribute(x)
 #  define magick_unused(x)  magick_unused_ ## x __attribute__((unused))
 #elif defined(MAGICKCORE_WINDOWS_SUPPORT) && !defined(__CYGWIN__)
 #  define magick_aligned(x)  __declspec(align(x))
